src/components/GroupSystems/GroupImmutableSystems.js

```diff
@@ -206,42 +207,9 @@ const GroupImmutableSystems = ({ groupName, groupId, ...props }) => {
     }
   }, [deviceData, selected, deviceImageSet]);
 
-  return (
-    <div id="group-systems-table">
-      {addToGroupModalOpen && (
-        <AddSystemsToGroupModal
-          isModalOpen={addToGroupModalOpen}
-          setIsModalOpen={(value) => {
-            dispatch(clearEntitiesAction());
-            setAddToGroupModalOpen(value);
-          }}
-          groupId={groupId}
-          groupName={groupName}
-          edgeParityIsAllowed={true}
-          activeTab={hybridInventoryTabKeys.immutable.key}
-        />
-      )}
-      {removeHostsFromGroupModalOpen && (
-        <RemoveHostsFromGroupModal
-          isModalOpen={removeHostsFromGroupModalOpen}
-          setIsModalOpen={setRemoveHostsFromGroupModalOpen}
-          modalState={currentSystem}
-        />
-      )}
-      {updateDevice && (
-        <AsyncComponent
-          appName="edge"
-          module="./UpdateDeviceModal"
-          navigateProp={useNavigate}
-          locationProp={useLocation}
-          notificationProp={notificationProp}
-          paramsProp={useParams}
-          updateModal={updateModal}
-          setUpdateModal={setUpdateModal}
-          refreshTable={() => true}
-        />
-      )}
-      {!addToGroupModalOpen && (
+  const InventoryTableWrapper = useMemo(
```

Comment:
Is there any specific reason to put it into a `useMemo`?

Reply:
It would also be better to put this into the `components/InventoryTable` and only wrap everything in a redux Provider if a `store` prop is passed.
Similar to what the `AsyncInventory` does: https://github.com/RedHatInsights/insights-inventory-frontend/blob/master/src/modules/AsyncInventory.js#L24

Reply:
> Is there any specific reason to put it into a `useMemo` ?

I have put this in a `useMemo` because without it, this value would get calculated on every render of the component which would create another instance of the store

Reply:
@LiorKGOW I think in this case you'd want to put the result of `getStore()` into a `useMemo`, but not the whole component.

Reply:
I have changed the PR according to this comment 
Thank you @bastilian !